Added create endpoint

madams51703 opened the pull request:
I want to add in some labels, but had to remove them to get the merge conflicts resolved.  Don't really understand why that generated a conflict, but for now I removed them.  After this merges, I want to add the labels back in and try another pull request.  I appreciate the patience with me as I learn more about git.

tuxmania87 commented:
Could you also remove all the trailing whitespace you added?

KvanTTT commented:
* Fix grammar for [test](https://travis-ci.org/antlr/grammars-v4/builds/255915565#L4728) passing.
* Add code fragment to test if possible.

madams51703 commented:
I removed whitespace from function_call rules, added example files and fixed the errors I introduced to the grammar.  Please let me know if I missed anything or if there is anything else I need to do.  Thanks again for your patience.

KvanTTT commented:
Travis [still failing](https://travis-ci.org/antlr/grammars-v4/builds/256026924#L4759).

madams51703 commented:
I will investigate further after work.

madams51703 commented:
Ok
 for i in examples/*.sql; do echo ${i}; cat ${i} | tr [:lower:] [:upper:] | grun TSql tsql_file ; done
examples/analytic_windowed_functions.sql
examples/control_flow.sql
examples/cursors.sql
examples/dbcc.sql
examples/ddl_alter_database_mirroring.sql
examples/ddl_create_alter_database.sql
examples/ddl_create_drop_type.sql
examples/ddl_create_endpoint.sql
examples/ddl_create_table.sql
examples/ddl_function.sql
examples/ddl_index.sql
examples/ddl_table.sql
examples/dml_delete.sql
examples/dml_insert.sql
examples/dml_merge.sql
examples/dml_openrowset.sql
examples/dml_select.sql
examples/dml_update.sql
examples/expressions.sql
examples/full_width_chars.sql
examples/predicates.sql
examples/procedures.sql
examples/statements.sql
examples/transactions.sql
examples/triggers.sql
examples/xml_data_type.sql
results of my testing against the examples

madams51703 commented:
I also added back the labels for the functions.  Those can be very useful in the future.

teverett commented:
@KvanTTT all good w/ this one?

KvanTTT commented on tsql/TSqlParser.g4:
```diff
@@ -1411,52 +1550,52 @@ derived_table
+    | ISNULL '(' expression ',' expression ')'		#ISNULL
     // https://docs.microsoft.com/en-us/sql/t-sql/xml/xml-data-type-methods
-    | xml_data_type_methods
+    | xml_data_type_methods				#XML_DATA_TYPE_METHODS
```
I'm not sure about alternative labels here. @Muhamedali what do you say about that?

madams51703 commented:
The reason for the labels is to be able to directly manipulate those
specific rules and provide translations for those functions

On Jul 22, 2017 5:14 PM, "Ivan Kochurkin" <notifications@github.com> wrote:

> *@KvanTTT* commented on this pull request.
> ------------------------------
>
> In tsql/TSqlParser.g4
> <https://github.com/antlr/grammars-v4/pull/808#discussion_r128905198>:
>
> >      // https://docs.microsoft.com/en-us/sql/t-sql/xml/xml-data-type-methods
> -    | xml_data_type_methods
> +    | xml_data_type_methods				#XML_DATA_TYPE_METHODS
>
> I'm not sure about alternative labels here. @Muhamedali
> <https://github.com/muhamedali> what do you say about that?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/antlr/grammars-v4/pull/808#pullrequestreview-51630892>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AcrrP_ZR3Fsp8cjxZ9bC2fhIvecA12XWks5sQmYdgaJpZM4Oe7-d>
> .
>

teverett commented:
@madams51703 @KvanTTT so, good to go?

madams51703 commented:
@teverett  I am good to go.  But I am biased.  :+)

KvanTTT commented:
Overall ok :+1:

madams51703 commented:
Oh, @KvanTTT  I read your comment from a different perspective.  If you specifically are referring to the xml_data_type_methods line, adding the labels is an all or nothing situation and I had no choice but to add it to the xml_data_type_methods line if I wanted to add it to the other functions in that ruleset.
